Mathlib/Algebra/DirectLimit.lean

```diff
@@ -172,6 +173,58 @@ theorem lift_unique [IsDirected ι (· ≤ ·)] (F : DirectLimit G f →ₗ[R] P
   · exact DirectLimit.induction_on x fun i x => by rw [lift_of]; rfl
 #align module.direct_limit.lift_unique Module.DirectLimit.lift_unique
 
+lemma lift_injective [IsDirected ι (· ≤ ·)]
+    (injective : ∀ i, Function.Injective <| g i) :
+    Function.Injective (lift R ι G f g Hg) := by
+  cases isEmpty_or_nonempty ι
+  · apply Function.injective_of_subsingleton
+  simp_rw [injective_iff_map_eq_zero] at injective ⊢
+  intros z hz
+  induction' z using DirectLimit.induction_on with _ g
+  rw [lift_of] at hz
+  rw [injective _ g hz, _root_.map_zero]
+
+variable (f) in
+/--
+For two family of modules `G` and `G'` indexed by the same set with direct system `f` and `f'`
+respectively, if there is a compatible family of linear-equivalence `eᵢ : Gᵢ ≅ Gᵢ'`, i.e. for
+`i ≤ j`, we have `eᵢ ∘ fᵢⱼ = fᵢⱼ' ∘ eⱼ`, there is a linear equivalence `lim G ≅ lim G'`.
+-/
+def congr [IsDirected ι (· ≤ ·)] (equiv : (i : ι) → G i ≃ₗ[R] G' i)
+    (compatible_equiv : ∀ i j h, (equiv j) ∘ₗ f i j h = f' i j h ∘ₗ equiv i) :
+    (DirectLimit G f) ≃ₗ[R] DirectLimit G' f' :=
+  LinearEquiv.ofLinear
+    (lift _ _ _ _ (fun i ↦ of _ _ _ _ _ ∘ₗ (equiv i).toLinearMap) fun i j h g ↦ by
+      have eq1 := FunLike.congr_fun (compatible_equiv i j h) g
+      simp only [LinearMap.coe_comp, LinearEquiv.coe_coe, Function.comp_apply] at eq1 ⊢
+      rw [eq1, of_f])
+    (lift _ _ _ _ (fun i ↦ of _ _ _ _ _ ∘ₗ (equiv i).symm.toLinearMap) fun i j h g ↦ by
+      have eq1 := FunLike.congr_arg (equiv _).symm <| FunLike.congr_fun (compatible_equiv i j h) <|
+        (equiv i).symm g
+      simp only [LinearMap.coe_comp, LinearEquiv.coe_coe, Function.comp_apply,
+        LinearEquiv.symm_apply_apply, LinearEquiv.apply_symm_apply] at eq1 ⊢
+      rw [← eq1, of_f])
+    (FunLike.ext _ _ fun x ↦ (isEmpty_or_nonempty ι).elim (fun _ ↦ Subsingleton.elim _ _)
+      fun _ ↦ DirectLimit.induction_on x fun _ _ ↦ by simp [lift_of])
+    (FunLike.ext _ _ fun x ↦ (isEmpty_or_nonempty ι).elim (fun _ ↦ Subsingleton.elim _ _)
+      fun _ ↦ DirectLimit.induction_on x fun _ _ ↦ by simp [lift_of])
+
+@[simp] lemma congr_apply_of [IsDirected ι (· ≤ ·)]
+    (equiv : (i : ι) → G i ≃ₗ[R] G' i)
+    (compatible_equiv : ∀ i j h, (equiv j) ∘ₗ f i j h = f' i j h ∘ₗ equiv i)
+    {i : ι} (g : G i) :
+    congr f f' equiv compatible_equiv (of _ _ _ _ i g : DirectLimit G f) =
+    of R ι G' f' i (equiv i g) := by
+  simpa [congr] using lift_of _ _ _
+
+@[simp] lemma congr_symm_apply_of [IsDirected ι (· ≤ ·)]
+    (equiv : (i : ι) → G i ≃ₗ[R] G' i)
+    (compatible_equiv : ∀ i j h, (equiv j) ∘ₗ f i j h = f' i j h ∘ₗ equiv i)
+    {i : ι} (g : G' i) :
+    (congr f f' equiv compatible_equiv).symm (of _ _ _ _ i g) =
+    of R ι G f i ((equiv i).symm g) := by
+  simpa [congr] using lift_of _ _ _
```

Comment:
These lemmas seem to be missing for the `Ring` and `AddCommGroup` cases.

Reply:
indeed, they are added now. Thanks!